packages/shared/src/hooks/squads/useSquadPostModeration.ts

```diff
@@ -0,0 +1,114 @@
+import { useMutation } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import {
+  PostModerationReason,
+  SquadPostRejectionProps,
+  squadApproveMutation,
+  squadRejectMutation,
+} from '../../graphql/squads';
+import { useLazyModal } from '../useLazyModal';
+import { LazyModal } from '../../components/modals/common/types';
+import { usePrompt } from '../usePrompt';
+import { useToastNotification } from '../useToastNotification';
+
+export const rejectReasons: { value: PostModerationReason; label: string }[] = [
+  {
+    value: PostModerationReason.OffTopic,
+    label: 'Off-topic post unrelated to the squad’s purpose',
+  },
+  {
+    value: PostModerationReason.Violation,
+    label: 'Violates the squad’s code of conduct',
+  },
+  {
+    value: PostModerationReason.Promotional,
+    label: 'Too promotional without adding value to the discussion',
+  },
+  {
+    value: PostModerationReason.Duplicate,
+    label: 'Duplicate or similar content already posted',
+  },
+  { value: PostModerationReason.LowQuality, label: 'Lacks quality or clarity' },
+  {
+    value: PostModerationReason.NSFW,
+    label: 'Contains inappropriate, NSFW or offensive content',
+  },
+  { value: PostModerationReason.Spam, label: 'Post is spam or scam' },
+  {
+    value: PostModerationReason.Misinformation,
+    label: 'Contains misleading or false information',
+  },
+  {
+    value: PostModerationReason.Copyright,
+    label: 'Copyright or privacy violation',
+  },
+  { value: PostModerationReason.Other, label: 'Other' },
+];
+
+interface UseSquadPostModeration {
+  onApprove: (ids: string[]) => Promise<void>;
+  onReject: (id: string) => void;
+  isLoading: boolean;
+  isSuccess: boolean;
+}
```

Comment:
This is basically the `CommonModerationProps` from the RFC, just had to rename the name to follow the hook's naming convention.